### PR TITLE
feat: Push notifications on follow request

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Companion PR here: https://github.com/ApollosProject/apollos-templates/pull/296

This is a fun one ... 

Adds in a check for the `actions` that OneSignal sends over, and then lets users provide a custom function to fire when an action matches a passed prop. You can see that being used here: 
https://github.com/ApollosProject/apollos-templates/pull/296/files#diff-9b1509d1d4517465bef7cbb0733f3cf179f1e9b555e50041214d2103218846e6R21-R28

Also adds a call to onesignal to send a notification when a user requests to follow another user. That part is straightforward.

### How do I test this PR?

This is what I did.

1. Clone this PR and it's companion. you need to have a database_url and run postgres.
2. Triple check your env variables. You should have an OneSignal api key and id in your api, and a matching one signal id in your client. 
3. Boot android locally. Check in onesignal that your device showed up. If it doesn't, double check the names of your API keys. 
4. Using graphql, send a follow request to whoever is logged into your android sim. You'll have to use the postgres server to find your apollos id. 
5. Accept that follow request in the sim! 

Note! If you view the notification and then dismiss it, or pull down poorly on the top bar and half view the notification, the button goes away. It's part of the android ui, and it's less of a problem on a real device where it's easy to pull the top bar up and down.

### CODE REVIEW

**Comment by redreceipt** [comment]
This is awesome! We're blowing past Rock's capabilities with this PR. I'll test soon.

File: packages/apollos-ui-notifications/src/Provider.js [hunk 1]
```diff
@@ -98,6 +103,13 @@ class NotificationsInit extends Component {
     const url = get(openResult, 'notification.payload.additionalData.url');
     if (url) {
       this.navigate(url);
+    } else if (
```
Comment (conrad-vanl) [inline packages/apollos-ui-notifications/src/Provider.js:106]: Should this be an `else if` or just a separate `if` ? Is there a scenario where we'd have a push notification with an action that also opens a URL?

**Review by conrad-vanl — APPROVED** [review]
Flow makes sense and is pretty simple. One question but good to merge

File: packages/apollos-ui-notifications/src/Provider.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-ui-notifications/src/Provider.js:106]: Good callout. I think we actually want the logic as it is, but reversed. Look for an action first, if we don't have one, check for a URL.

Comment (vinnyjth) [reply packages/apollos-ui-notifications/src/Provider.js]: That way we can use `urls` as fallbacks for buttons with no associated logic deployed (like older apps)

**Pull request merged**